.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,40 +1,63 @@
 ## PR Summary
 
-## PR Checklist
+<!--
+Thank you so much for your PR! To help us review your contribution, please
+summarize the purpose of this PR here.
 
-- [ ] Has Pytest style unit tests
-- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
-- [ ] New features are documented, with examples if plot related
-- [ ] Documentation is sphinx and numpydoc compliant
-- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
-- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way
+- The summary should provide at least 1-2 sentences describing the pull request
+  in detail (Why is this change required? What problem does it solve?) and
+  link to any relevant issues.
+-->
 
-<!--
-Thank you so much for your PR!  To help us review your contribution, please
-consider the following points:
 
-- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.
+## PR Checklists
 
-- Help with git and github is available at
-  https://matplotlib.org/devel/gitwash/development_workflow.html.
+<!-- Feel free to delete any checklists that do not apply to this PR -->
+
+For new code:
+
+- [ ] Has Pytest style unit tests (and `pytest lib/matplotlib/tests` passes)
+- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check)
+- [ ] New code is documented, with examples if plot related
+
+For new documentation:
+
+- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error)
+- [ ] New documentation conforms to matplotlib style conventions. (If you have `flake8-docstrings` and `pydocstyle<4` installed, run `flake8 --docstring-convention=all` on changed files to check).
+
+For new features:
+
+- [ ] Added an entry to `doc/users/next_whats_new/` if major new feature (follow instructions in `README.rst` there)
```

Comment:
I think the there is ambigous and explicit whats_new/readme.rst might be clearer